lib/shopify-cli/commands/deploy.rb

```diff
@@ -4,7 +4,9 @@
 
 module ShopifyCli
   module Commands
-    class Deploy < ShopifyCli::Command
+    class Deploy < ShopifyCli::ContextualCommand
+      available_in :app
+
```

Comment:
This method can be invoked multiple times for commands that exist in multiple contexts, like:
```
available_in :app 
available_in :script
```
Although it's likely the implementation will be different so it would probably be better to implement them as separate classes that share the same command_name.

Reply:
For this, I would use the same pattern as established in #406. One `Deploy` class but use `app_type` (probably should be renamed to be more generic) to point to different implementations based on the metadata in the yml file.

Reply:
Are you planning on pulling the relevant changes from #406 into this PR?  I am wondering at which point is a good time for me to integrate Scripts?  Is this PR complete for what we discussed in [context awareness ticket](https://github.com/Shopify/script-service/issues/349)?